include/vcpkg/base/expected.h

```diff
@@ -198,25 +200,25 @@ namespace vcpkg
             }
         }
 
-        template<class F, class U = map_t<F>>
-        U then(F f) const&
+        template<class F, class... Args, class U = std::invoke_result_t<F, const_ref_type, Args&&...>>
```

Comment:
Why don't we just use:

```c++
template <class F, class... Args>
std::invoke_result_t<F, const_ref_type, Args&&...> then(F f, Args&&... args) const&;
```

Reply:
I think I originally had some `decltype`s in there that needed the arguments. Will fix.